tooling/debugger/src/repl.rs

```diff
@@ -121,9 +137,76 @@ impl<'a, B: BlackBoxFunctionSolver> ReplDebugger<'a, B> {
         }
     }
 
+    fn display_opcodes(&self) {
+        let opcodes = self.context.get_opcodes();
+        let current_opcode_location = self.context.get_current_opcode_location();
+        let current_ip = match current_opcode_location {
+            Some(OpcodeLocation::Acir(ip)) => Some(ip),
+            Some(OpcodeLocation::Brillig { acir_index, .. }) => Some(acir_index),
+            None => None,
+        };
+        let current_brillig_pc = match current_opcode_location {
+            Some(OpcodeLocation::Brillig { brillig_index, .. }) => brillig_index,
+            _ => 0,
+        };
+        for (ip, opcode) in opcodes.iter().enumerate() {
+            let marker = if current_ip == Some(ip) {
+                "->"
+            } else if self.context.is_breakpoint_set(&OpcodeLocation::Acir(ip)) {
+                " *"
+            } else {
+                ""
+            };
+            if let Opcode::Brillig(brillig) = opcode {
+                println!("{:>3} {:2} BRILLIG inputs={:?}", ip, marker, brillig.inputs);
+                println!("       |       outputs={:?}", brillig.outputs);
+                for (pc, brillig_opcode) in brillig.bytecode.iter().enumerate() {
+                    println!(
+                        "{:>3}.{:<2} |{:2} {:?}",
+                        ip,
+                        pc,
+                        if pc == current_brillig_pc {
+                            marker
+                        } else if self.context.is_breakpoint_set(&OpcodeLocation::Brillig {
+                            acir_index: ip,
+                            brillig_index: pc,
+                        }) {
+                            " *"
+                        } else {
+                            ""
+                        },
+                        brillig_opcode
+                    );
+                }
+            } else {
+                println!("{:>3} {:2} {:?}", ip, marker, opcode);
+            }
+        }
+    }
+
+    fn add_breakpoint_at(&mut self, location: OpcodeLocation) {
+        if self.context.is_breakpoint_set(&location) {
+            println!("Breakpoint at opcode {location} already set");
+        } else if !self.context.is_valid_location(&location) {
+            println!("Invalid opcode location {location}");
+        } else {
+            self.context.add_breakpoint(location);
+            println!("Added breakpoint at opcode {location}");
+        }
+    }
+
+    fn delete_breakpoint_at(&mut self, location: OpcodeLocation) {
+        if self.context.is_breakpoint_set(&location) {
+            self.context.delete_breakpoint(&location);
+            println!("Breakpoint at opcode {location} deleted");
+        } else {
+            println!("Breakpoint at opcode {location} not set");
+        }
+    }
```

Comment:
Let's pass back out the boolean result from the `insert` and `remove` on the `HashSet` which will allow us to simplify these functions.